src/main/java/uk/gov/service/notify/Notification.java

```diff
@@ -3,32 +3,35 @@
 import org.joda.time.DateTime;
 import org.json.JSONObject;
 
+import java.util.Optional;
+import java.util.UUID;
+
 public class Notification {
-    private String id;
-    private String body;
-    private int contentCharCount;
+    private UUID id;
+    private Optional<String> reference;
```

Comment:
Optional is frowned upon as a field or method parameter usually.
See [stackoverflow](http://stackoverflow.com/a/39005452/4951015) for some discussion on this.
Jetbrains IDEs will recommend against using it.
Optional isn't serialisable.
Its gets very verbose if there's lots of optional elements